File: runs/PlotUtils/Population.py
import numpy as np

class Popultion:
    def __init__(self):
        self.all = dict()
        self.size = 0

    def keys(self):
        return self.all.keys()

    def label(self, subpop):
        b = self.__binarize(subpop)

        if b not in self.keys():
            self.size += 1
            self.all[b] = self.size 

        return self.all[b]

    def __binarize(self, subpop):
        subpop_char_list = [str(int(s)) for s in subpop]
        return ''.join(subpop_char_list)

# Example usage:
# pop = Popultion()
# pop.label([1,0,1])  # 1
# pop.label([1,0,1])  # 1
# pop.label([1,1,1])  # 2
# pop.label([1,1,1,1])  # 3

File: runs/PlotUtils/Population.jl
mutable struct Population
  all::Dict{String, Int}
  size::Int
end

Population() = Population(Dict(), 0)
Base.keys(p::Population) = keys(p.all)

function binarize(p::Population, subpop)
  subpop_char_list = [string(Int(s)) for s in subpop]
  return join(subpop_char_list, "")
end

function label(p::Population, subpop)
  b = binarize(p, subpop)
  if !(b in keys(p))
    p.size += 1
    p.all[b] = p.size
  end
  return p.all[b]
end

# Example usage:
# p = Population()
# label(p, [1,0,1])  # 1
# label(p, [1,0,1])  # 1
# label(p, [1,1,1])  # 2
# label(p, [1,1,1,1]) # 3
